src/Avalonia.Base/Styling/ChildSelector.cs

```diff
@@ -31,7 +31,9 @@ public override string ToString(Style? owner)
         {
             if (_selectorString == null)
             {
-                _selectorString = _parent.ToString(owner) + " > ";
+                _selectorString = _parent is OrSelector ?
+                    $"({_parent.ToString(owner)}) > " :
```

Comment:
Note: all selectors have been modified to add parenthesis around `OrSelector`, why not add this directly to `OrSelector.ToString()`? You could add an internal override taking a flag to avoid adding parenthesis around an `OrSelector` being placed at the root level.

Reply:
What I meant is to have a method like `internal virtual string ToString(Style owner, bool hasChild)` on `Selector` and just call that with `hasChild: true` in every child selector without checking for the parent. Then override it in `OrSelector` to add the parenthesis according to `hasChild`.